### PR TITLE
[NO-ISSUE] fix/clippy warnings

### PR DESCRIPTION
**Pull request by GuillaumeDecMeetsMore**
### Changed
- Fix some Clippy warnings
  - For most of them (95%), I directly applied the suggested change in VSCode (in other words, these have been automatically fixed)

### Notes
- In Rust, there is `cargo check` (a default command of Cargo, the npm of Rust). This is like a simple "eslint"
- And there is [Clippy](https://github.com/rust-lang/rust-clippy), which is more like the "tslint" of Rust - it does more advanced checks based on types
  - Clippy is also developed by the contributors working on Rust

### CODE REVIEW

File: scheduler/clients/rust/src/service.rs [hunk 1]
```diff
@@ -82,9 +81,6 @@ pub struct CreateServiceInput {
     pub multi_person: Option<ServiceMultiPersonOptions>,
 }
 
-#[derive(Serialize)]
-struct Empty {}
```
Comment (GuillaumeDecMeetsMore) [inline scheduler/clients/rust/src/service.rs:86]: Unused

File: scheduler/clients/rust/src/user.rs [hunk 2]
```diff
@@ -89,7 +89,7 @@ impl UserClient {
         let query: String = input.into();
         self.base
             .get(
-                format!("user/{}/freebusy{}", user_id.to_string(), query),
```
Comment (GuillaumeDecMeetsMore) [inline scheduler/clients/rust/src/user.rs:92]: `to_string()` is automatically called by `format!` when needed

File: scheduler/crates/api/src/event/sync_event_reminders.rs [hunk 3]
```diff
@@ -114,7 +114,6 @@ async fn create_event_reminders(
                         })
                         .collect::<Vec<_>>()
                 })
-                .flatten()
```
Comment (GuillaumeDecMeetsMore) [inline scheduler/crates/api/src/event/sync_event_reminders.rs:117]: merge `map + flatten` into `flat_map`

File: scheduler/crates/api/src/event/update_event.rs [hunk 4]
```diff
@@ -147,10 +147,10 @@ impl UseCase for UpdateEventUseCase {
             }
         };
 
-        e.service_id = service_id.clone();
```
Comment (GuillaumeDecMeetsMore) [inline scheduler/crates/api/src/event/update_event.rs:150]: https://rust-lang.github.io/rust-clippy/master/index.html#/assigning_clones

File: scheduler/crates/api/src/lib.rs [hunk 5]
```diff
@@ -78,7 +78,7 @@ impl Application {
     async fn configure_server(context: NettuContext) -> Result<(Server, u16), std::io::Error> {
         let port = context.config.port;
         let address = format!("127.0.0.1:{}", port);
-        let listener = TcpListener::bind(&address)?;
+        let listener = TcpListener::bind(address)?;
```
Comment (GuillaumeDecMeetsMore) [inline scheduler/crates/api/src/lib.rs:81]: No need to use a reference, `address` will be converted automatically if needed
https://rust-lang.github.io/rust-clippy/master/index.html#needless_borrows_for_generic_args

File: scheduler/crates/api/src/shared/auth/policy.rs [hunk 6]
```diff
@@ -76,77 +76,68 @@ mod test {
     fn permissions() {
         let policy = Policy::default();
         assert!(policy.authorize(&Vec::new()));
-        assert!(!policy.authorize(&vec![Permission::CreateCalendar]));
+        assert!(!policy.authorize(&[Permission::CreateCalendar]));
```
Comment (GuillaumeDecMeetsMore) [inline scheduler/crates/api/src/shared/auth/policy.rs:79]: No need to use a Vector
https://rust-lang.github.io/rust-clippy/master/index.html#/useless_vec

File: scheduler/crates/api/src/shared/auth/route_guards.rs [hunk 7]
```diff
@@ -177,7 +177,7 @@ pub async fn protect_public_account_route(
             }
         }
         // No nettu-account header, then check if this is an admin client
-        None => return protect_account_route(http_req, ctx).await,
+        None => protect_account_route(http_req, ctx).await,
```
Comment (GuillaumeDecMeetsMore) [inline scheduler/crates/api/src/shared/auth/route_guards.rs:180]: `return` is implicit
https://rust-lang.github.io/rust-clippy/master/index.html#/needless_return

File: scheduler/crates/api/src/user/get_user_freebusy.rs [hunk 8]
```diff
@@ -12,9 +12,7 @@ pub fn parse_vec_query_value(val: &Option<String>) -> Option<Vec<ID>> {
     val.as_ref().map(|ids| {
         ids.split(',')
             .map(String::from)
-            .map(|id| id.parse::<ID>())
-            .filter(|id| id.is_ok())
-            .map(|id| id.unwrap())
+            .flat_map(|id| id.parse::<ID>())
```
Comment (GuillaumeDecMeetsMore) [inline scheduler/crates/api/src/user/get_user_freebusy.rs:15]: https://rust-lang.github.io/rust-clippy/master/index.html#/result_filter_map

File: scheduler/crates/api/src/user/get_user_freebusy.rs [hunk 9]
```diff
@@ -121,10 +119,7 @@ impl GetFreeBusyUseCase {
         let mut calendars = ctx.repos.calendars.find_by_user(&self.user_id).await;
 
         if !calendar_ids.is_empty() {
-            calendars = calendars
-                .into_iter()
-                .filter(|cal| calendar_ids.contains(&cal.id))
-                .collect();
+            calendars.retain(|cal| calendar_ids.contains(&cal.id));
```
Comment (GuillaumeDecMeetsMore) [inline scheduler/crates/api/src/user/get_user_freebusy.rs:122]: https://rust-lang.github.io/rust-clippy/master/index.html#/manual_retain

File: scheduler/crates/domain/src/booking_slots.rs [hunk 10]
```diff
@@ -18,12 +18,10 @@ fn is_cursor_in_events(
...
-    for event in events.as_ref() {
-        if event.start_ts <= cursor && event.end_ts >= cursor + duration {
-            return Some(event);
-        }
-    }
-    None
+    events
+        .as_ref()
+        .iter()
+        .find(|&event| event.start_ts <= cursor && event.end_ts >= cursor + duration)
```
Comment (GuillaumeDecMeetsMore) [inline scheduler/crates/domain/src/booking_slots.rs:24]: https://rust-lang.github.io/rust-clippy/master/index.html#/manual_find

File: scheduler/crates/domain/src/schedule.rs [hunk 11]
```diff
@@ -226,11 +226,11 @@ impl ScheduleRule {
         let mut remove_intervals = HashMap::new();
 
         for i in 0..self.intervals.len() {
-            if remove_intervals.get(&i).is_some() {
+            if remove_intervals.contains_key(&i) {
```
Comment (GuillaumeDecMeetsMore) [inline scheduler/crates/domain/src/schedule.rs:229]: https://rust-lang.github.io/rust-clippy/master/index.html#/unnecessary_get_then_check

File: scheduler/crates/domain/src/user.rs [hunk 12]
```diff
@@ -45,19 +45,14 @@ pub struct UserIntegration {
...
     Google,
     Outlook,
 }
 
-impl Default for IntegrationProvider {
-    fn default() -> Self {
-        IntegrationProvider::Google
-    }
-}
-
```
Comment (GuillaumeDecMeetsMore) [inline scheduler/crates/domain/src/user.rs:60]: https://rust-lang.github.io/rust-clippy/master/index.html#/derivable_impls

File: scheduler/crates/infra/src/repos/account_integrations/mod.rs [hunk 13]
```diff
@@ -59,13 +59,10 @@ mod tests {
         assert_eq!(acc_integrations[1].account_id, account.id);
         assert!(acc_integrations
             .iter()
-            .find(|c| c.provider == IntegrationProvider::Google)
-            .is_some());
+            .any(|c| c.provider == IntegrationProvider::Google));
```
Comment (GuillaumeDecMeetsMore) [inline scheduler/crates/infra/src/repos/account_integrations/mod.rs:62]: https://rust-lang.github.io/rust-clippy/master/index.html#/search_is_some

File: scheduler/crates/infra/src/services/outlook_calendar/auth_provider.rs [hunk 14]
```diff
@@ -102,7 +102,6 @@ pub async fn exchange_code_token(req: CodeTokenRequest) -> Result<CodeTokenRespo
     let scopes = res
         .scope
         .split(' ')
-        .into_iter()
```
Comment (GuillaumeDecMeetsMore) [inline scheduler/crates/infra/src/services/outlook_calendar/auth_provider.rs:105]: https://rust-lang.github.io/rust-clippy/master/index.html#/useless_conversion

**Review by mm-zacharydavison — APPROVED** [review]
Educational PR!

**Pull request merged**